mock/py/mockbuild/util.py

```diff
@@ -1300,12 +1300,18 @@ def load_config(config_path, name, uidManager, version, pkg_python_dir):
         '~' + pwd.getpwuid(os.getuid())[0]), '.config/mock.cfg')
     do_update_config(log, config_opts, cfg, uidManager, name)
 
-    # default /etc/hosts contents
-    if not config_opts['use_host_resolv'] and 'etc/hosts' not in config_opts['files']:
-        config_opts['files']['etc/hosts'] = dedent('''\
-            127.0.0.1 localhost localhost.localdomain
-            ::1       localhost localhost.localdomain localhost6 localhost6.localdomain6
-            ''')
+    if not config_opts['use_host_resolv']:
+        # default /etc/hosts contents
+        if 'etc/hosts' not in config_opts['files']:
+            config_opts['files']['etc/hosts'] = dedent('''\
+                127.0.0.1 localhost localhost.localdomain
+                ::1       localhost localhost.localdomain localhost6 localhost6.localdomain6
+                ''')
+        # bind mount an empty /etc/resolv.conf when using nspawn and networking is disabled
+        if config_opts['use_nspawn'] and not config_opts['rpmbuild_networking']:
+            config_opts['files']['var/tmp/empty'] = ''
+            config_opts['nspawn_args'] += ['--bind=+/var/tmp/empty:/etc/resolv.conf']
```

Comment:
Can this be rather: `--bind=:/etc/resolv.conf'?
Quoting from man page:
```
or a colon-separated pair of paths — in which case the first
specified path is the source in the host, and the second path is the destination in the container,...
... 
The source path may be specified as empty string, in which case a temporary directory below the host's /var/tmp directory is used. It is
automatically removed when the container is shut down.
```

Reply:
I can try that.  Reading it, I thought it would bind mount a directory and that a dir as /etc/resolv.conf wouldn't work well.  But let's confirm that empirically. :)

Reply:
Sadly, that does fail:

```
Start: shell
Cannot bind mount directory /var/tmp/nspawn-temp-8SCtjW/src on file /var/lib/mock/fedora-27-x86_64/root/etc/resolv.conf.
```

That's with `config_opts['nspawn_args'] += ['--bind=:/etc/resolv.conf']`.  I scrubbed the root-cache and chroot, but perhaps there's still an /etc/resolv.conf present which causes the error.  I do still wonder how many things would not be happy if nspawn would bind mount a dir at /etc/resolv.conf.

We can keep at that if you think it's going to be cleaner than mounting an empty file from the host (at least on EL7).  We could use `+/var/tmp/empty` on Fedora and EL8, but fall back to a file on the host for EL7, perhaps.  I'm not sure it's worth the extra code to do that, but it's an option.